### PR TITLE
PLANET-2011 Added Media library setting page

### PR DESCRIPTION
**Pull request by sagarsdeshmukh**

### CODE REVIEW

File: includes/settings.twig [hunk 1]
```diff
@@ -2,5 +2,34 @@
 
     {% embed "message.twig" with messages %}{% endembed %}
 
+    <div id="settings_div" class="wrap">
+        <h2>{{ __( 'GP Media Library Settings', domain ) }}</h2>
+    </div>
+
+    {{ fn( 'settings_errors' ) }}
+
+    <form id="p4ml_main_settings_form" name="p4ml_main_settings_form" method="post" action="options.php"> {# ../../../../wp-admin/options.php #}
```
Comment (kirdia) [inline includes/settings.twig]: Please remove this comment.

File: includes/message.twig [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{% block message_content %}
+
+    {% if ( messages ) %}
```
Comment (kirdia) [inline includes/message.twig]: Indentation seems off here..

File: classes/controller/menu/class-settings-controller.php [hunk 3]
```diff
@@ -81,19 +81,19 @@ public function validate( $settings ): bool {
 			$has_errors = false;
```
Comment (angtheod) [inline classes/controller/menu/class-settings-controller.php]: missing space before ':'

File: classes/controller/menu/class-settings-controller.php [hunk 4]
```diff
@@ -116,14 +116,5 @@ public function sanitize( &$settings ) {
...
-		/**
-		 * Loads the saved language.
-		 */
-		public function set_locale(): string {
-			$main_settings = get_option( 'p4ml_main_settings' );
-
-			return isset( $main_settings['p4ml_lang'] ) ? $main_settings['p4ml_lang'] : '';
-		}
 	}
 }
```
Comment (angtheod) [inline classes/controller/menu/class-settings-controller.php]: please add new line at end of file

File: includes/settings.twig [hunk 5]
```diff
@@ -2,5 +2,34 @@
...
+                <td>
+                    <input id="p4ml_api_password" name="p4ml_main_settings[p4ml_api_password]" value="{{ settings.p4ml_api_password|e('html_attr') }}" size="50" placeholder="{{ __( 'Enter your GP media library account password', domain ) }}" type="password" />
+                </td>
+            </tr>
+        </table>
+        {{ fn( 'submit_button', __( 'Save Changes', domain ), 'primary', 'p4ml_main_settings_save_button' ) }}
+    </form>
+    <div class="clear"></div>
 
 {% endblock %}
```
Comment (angtheod) [inline includes/settings.twig]: please add new line at end of file

**Comment by angtheod** [comment]
Some comments on code not included in this PR but in previous one #1 

1. Remove commented code
https://github.com/greenpeace/planet4-plugin-medialibrary/pull/1/files#diff-f831146c1c5bd5a7b8efdc4d81370db0R59

2. Missing function doc comments (lines 66, 72, 77)
https://github.com/greenpeace/planet4-plugin-medialibrary/pull/1/files#diff-f831146c1c5bd5a7b8efdc4d81370db0R66

3. Need to escape html output
https://github.com/greenpeace/planet4-plugin-medialibrary/pull/1/files#diff-f831146c1c5bd5a7b8efdc4d81370db0R79

**Pull request merged**